design/FeaturesWithLimits.md

```diff
@@ -0,0 +1,119 @@
+# Features that add new limits
+
+Some new features add associated limits. This document records a design policy for such features.
+
+## Common behaviors
+
+- `{ newLimit: undefined }` is allowed even if `newLimit` is unknown. (It has no effect.)
+- Limits entries associated with features have these Adapter Capability Guarantees:
+  - If the feature is available, its limits are `better` than or equal to the default.
+  - If not, its limits are `undefined`.
+  - They are `undefined` if the feature is unavailable on the adapter.
+- It is allowed to request a limit that's available on the adapter (that is,
+  not `undefined` in `adapter.limits`), even without also requesting the corresponding feature.
+  It won't have any effect, but the limit request will still be validated against the adapter.
+- It is allowed to request a feature that's available on the adapter, even without also
+  requesting the corresponding limit(s). (They will get their default values.)
+
+## Policies for new features
+
+- There should be a feature name for each such feature - don't just add a new limit
+  (defaulting to 0) on its own.
+  - Allows us to clearly guarantee a default value for the limit when the feature is available.
+    (You only have to request the feature, not figure out what limit you need to request.)
+  - Is consistent with features that add 0 limits or more than 1 limit.
+- Each new limit should define its "default" value as the default non-trivial value
+  (that is, ignoring whether the feature is available/requested).
+
+## Example
+
+The new feature is `"foo"`.
+The new limit is `maxFoo`, and it has a default value of `5`.
+
+### Case Studies
+
+For each case study we'll look at how a piece of code behaves in several cases given an adapter `a`:
+
+- *Available (5)*: a browser which implements `"foo"`, on a device with support with `maxFoo: 5`.
+  - `a.features.has("foo")` is `true`
+  - `'maxFoo' in a.requiredLimits` is `true`
+  - `a.requiredLimits.maxFoo` is `7`
```

Comment:
Did you mean the following?
```suggestion
  - `'maxFoo' in a.limits` is `true`
  - `a.limits.maxFoo` is `7`
```
Similar would apply in the next 3 case studies.

Reply:
Yes, I think so. Thanks, will fix soon.